### ISSUE TITLE
Not prompted to login

### ISSUE THREAD

**Issue by jon-morris** [microsoftgraph/uwp-csharp-snippets-sample#7]
After successfully building and launching the app, I am not being prompted to log in, The page with all the Stories appears but I get an error when I try to run any of them. 

Have I got this right?:

![image](https://cloud.githubusercontent.com/assets/16494175/15560965/dc959b4c-22a5-11e6-8be3-3f48ff076720.png)

Is a certificate required?  I am getting the following warning: "Certificate file 'Microsoft-Graph-Snippets-SDK_TemporaryKey.pfx' not found."

Thanks!

**Comment by jon-morris** [comment]
Here are the errors.  BTW, I did register the app with the App Registration Portal as instructed.

'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: DefaultDomain): Loaded 'C:\Program Files\WindowsApps\Microsoft.NET.CoreRuntime.1.0_1.0.23819.0_x64__8wekyb3d8bbwe\mscorlib.ni.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\entrypoint\Microsoft-Graph-Snippets-SDK.exe'. Symbols loaded.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Runtime.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\WinMetadata\Windows.winmd'. Module was built without symbols.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Runtime.InteropServices.WindowsRuntime.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Diagnostics.Debug.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Runtime.WindowsRuntime.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Runtime.WindowsRuntime.UI.Xaml.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Collections.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.ObjectModel.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Threading.Tasks.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Private.Uri.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Linq.dll'. Cannot find or open the PDB file.
Oops - App not registered. To run this sample, you must register it with the App Registration Portal. See Readme for more info.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Threading.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\System.Runtime.Extensions.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\Microsoft.Graph.dll'. Cannot find or open the PDB file.
'Microsoft-Graph-Snippets-SDK.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Projects\uwp-csharp-snippets-sample-master\Microsoft-Graph-Snippets-SDK\bin\x64\Debug\AppX\Microsoft.Identity.Client.dll'. Cannot find or open the PDB file.
Exception thrown: 'System.Runtime.InteropServices.COMException' in mscorlib.ni.dll
Exception thrown: 'System.Runtime.InteropServices.COMException' in mscorlib.ni.dll
WinRT information: Cannot find a resource with the given key.
Exception thrown: 'System.TypeInitializationException' in Microsoft-Graph-Snippets-SDK.exe
Exception thrown: 'System.TypeInitializationException' in mscorlib.ni.dll
Exception thrown: 'System.TypeInitializationException' in mscorlib.ni.dll
Users.Get Events failed. Exception: The type initializer for 'Microsoft_Graph_Snippets_SDK.AuthenticationHelper' threw an exception.

**Comment by jamescro** [comment]
I downloaded this repo as a zip and registered a new client for it. I couldn't reproduce what you're seeing.
Did you add the mobile platform to the client in the app registration portal, and did you copy the client id and return URI values into the app.xaml file? The only way that I'm able to reproduce the behavior you're seeing is not to copy the client id and return URI values into the app.xaml file.
One other possibility is that you're running into a manifestation of the VS path length issue. Have you tried cloning or copying this repo into your c:// drive to shorten the path length for this solution?

**Comment by jamescro** [comment]
This error in your output indicates that the client id hasn't been added to the App.xaml file:
"Oops - App not registered. To run this sample, you must register it with the App Registration Portal. See Readme for more info."
 Did you place your client id here?

        
        <x:String x:Key="ida:ClientID"></x:String>

**Comment by jon-morris** [comment]
Yes, I registered the app via the App Registration Portal.  I placed the ClientID between the >< symbols in the above statement. Maybe that is incorrect? Exactly where in the tag should the ClientID be placed? Thanks!

**Comment by jon-morris** [comment]
I also tried x:Key="ida:123456789" but I got an error too.

**Comment by jamescro** [comment]
It should look like this: 
        
        <x:String x:Key="ida:ClientID">00c4caa5-bf18-4342-a16a-283fbd7d1dd9</x:String>

Could you put a break at this point in the MainPage.xaml.cs file?

        protected override void OnNavigatedTo(NavigationEventArgs e)
        {
            // Developer code - if you haven't registered the app yet, we warn you. 
            if (!App.Current.Resources.ContainsKey("ida:ClientID"))
            {
                Debug.WriteLine("Oops - App not registered with Office 365. To run this sample, you must register it with Office 365. See Readme for more info.");

            }
        }
That's where the "oops" output is coming from. Could you verify at that point that ContainsKey is returning false?

**Comment by jon-morris** [comment]
Jim,

Success!  I was not being prompted to login because when I registered my app, I logged into the App Registration Portal and created the app using my Office 365 tenant admin account.  Once I re-registered a new app using my Windows Live / Azure account and used the new ClientID generated for the new app, everything worked.

The "App Registration Portal" instructions in the readme say "Sign into the App Registration Portal using **either your personal or work or school account.** " but I incorrectly assumed that meant the Office 365 tenant admin account.  It makes sense to me now but not before I had run the app successfully.

Thanks for your help!

Jon
